files/solr_check_instance.sh

```diff
@@ -0,0 +1,44 @@
+#!/bin/bash -eu
+# Step through all cores under one instance and throw an exit code if something is not running.
+# Exit code 5: hard fail, no need to wait any more
+# Exit code 2: possibly not loaded yet, try waiting a bit
+
+set -o pipefail
+
+if [ -z "$1" ]; then
+	echo "Cowardly refusing to reload all cores on this server. Please specify an instance as the first param."
+	exit 2
+else
+	instance="${1}v4"
+	target="*/${instance}/*"
+fi
+
+status=$(env -i curl "http://localhost:8080/$instance/admin/cores?wt=json" -sS)
+for core in $(find /var/lib/solr -maxdepth 2 -mindepth 2 -path "$target" -type d -not -name logs); do
+	if [[ -f "$core/core.properties" ]]; then
```

Comment:
possibly do something like this to minimise the indent
```suggestion
	if [[ ! -f "${core}/core.properties" ]]; then continue; fi
```